sessionpeermanager/sessionpeermanager.go

```diff
@@ -101,9 +107,19 @@ func (spm *SessionPeerManager) FindMorePeers(ctx context.Context, c cid.Cid) {
 		// - manage timeouts
 		// - ensure two 'findprovs' calls for the same block don't run concurrently
 		// - share peers between sessions based on interest set
+		wg := &sync.WaitGroup{}
 		for p := range spm.network.FindProvidersAsync(ctx, k, 10) {
-			spm.peerMessages <- &peerFoundMessage{p}
+			wg.Add(1)
+			go func(p peer.ID) {
+				defer wg.Done()
+				err := spm.network.ConnectTo(ctx, p)
+				if err != nil {
+					log.Debugf("failed to connect to provider %s: %s", p, err)
+				}
+				spm.peerMessages <- &peerFoundMessage{p}
+			}(p)
 		}
+		wg.Wait()
```

Comment:
Nit: We don't really need to wait here.

Reply:
true I think... though we may need to once we dedup with providers worker. But agreed, maybe I'll remove wait group knowledge for now.

Reply:
Doesn't really matter.